src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -294,14 +294,9 @@ def deploy_nsd_from_bicep(
             os.path.join(self.config.build_output_folder_name, NF_DEFINITION_BICEP_FILE)
         )
 
-        arm_template_artifact_path = os.path.join(
-            self.config.build_output_folder_name, NF_DEFINITION_JSON_FILE
-        )
-
-        with open(arm_template_artifact_path, "w") as file:
+        with open(self.config.arm_template.file_path, "w") as file:
             file.write(json.dumps(arm_template_artifact_json, indent=4))
```

Comment:
I dont think you need the file.write, i think you do this:
        with open(mapping_file_path, "w", encoding="utf-8") as file:
            json.dump(data, file, indent=4)